src/controller/model/endpoint.ts

```diff
@@ -277,21 +277,31 @@ class Endpoint extends Entity {
                 options.disableResponse, options.disableRecovery, options.srcEndpoint) as Promise<Type>;
         }): Promise<Type> {
         const logPrefix = `Request Queue (${this.deviceIeeeAddress}/${this.ID}): `;
-        const request = new Request(func, frame, this.getDevice().pendingRequestTimeout, options.sendWhen,
-            options.sendPolicy);
+
+        if(options.sendWhen) {
```

Comment:
Could we also remove this part? (maybe in a next PR?)

Reply:
Absolutely! And yes, that will be another PR:-)

Reply:
Many thanks for your dedication 😄